device/transport/amqp/src/amqp.ts

```diff
@@ -244,7 +475,11 @@ export class Amqp extends EventEmitter implements Client.Transport, StableConnec
    */
   /*Codes_SRS_NODE_DEVICE_AMQP_16_013: [The ‘complete’ method shall call the ‘complete’ method of the C2D `ReceiverLink` object and pass it the message and the callback given as parameters.] */
   complete(message: Message, done?: (err?: Error, result?: results.MessageCompleted) => void): void {
```

Comment:
>complete [](start = 2, length = 8)

now that I'm completely out of control on the whole concurrency thing, should 'complete' be a FSM action and completingMessage be it's own state?  I think no, but I also wonder how bulletproof we want this to be.

Reply:
same for reject and abandon.

---
In reply to: [141136795](https://github.com/Azure/azure-iot-sdk-node/pull/89#discussion_r141136795) [](ancestors = 141136795)

Reply:
these are actions that are handled by the c2dlink state machine. 

---
In reply to: [141136795](https://github.com/Azure/azure-iot-sdk-node/pull/89#discussion_r141136795) [](ancestors = 141136795)